frontend/pages/policies/ManagePoliciesPage/components/AddPolicyModal/AddPolicyModal.tsx

```diff
@@ -0,0 +1,98 @@
+/* This component is used for creating policies */
+
+import React, { useState, useCallback } from "react";
+// @ts-ignore
+import Modal from "components/modals/Modal";
+import Button from "components/buttons/Button";
+import InfoBanner from "components/InfoBanner/InfoBanner";
+// @ts-ignore
+import Dropdown from "components/forms/fields/Dropdown";
+import { IQuery } from "interfaces/query";
+
+const baseClass = "add-policy-modal";
+
+interface IAddPolicyModalProps {
+  allQueries: IQuery[];
+  onCancel: () => void;
+  onSubmit: (query_id: any) => void;
+}
+
+const AddPolicyModal = ({
+  onCancel,
+  onSubmit,
+  allQueries,
+}: IAddPolicyModalProps): JSX.Element => {
+  const [selectedQuery, setSelectedQuery] = useState<number | undefined>();
+
+  const createQueryDropdownOptions = () => {
+    const queryOptions = allQueries.map((q) => {
+      return {
+        value: q.id,
+        label: q.name,
+      };
+    });
+    return queryOptions;
+  };
+
+  const onChangeSelectQuery = useCallback(
+    (queryId: number) => {
+      setSelectedQuery(queryId);
+    },
+    [setSelectedQuery]
+  );
+
```

Comment:
I'm probably missing a more elegant solution, but when I try to type `query_id` at line 21 as `number` I get a TypeScript error here if I don't handle the `undefined` case because up the chain the `create` method on `services/entites/policies` param is typed to expect `number`.